base/deprecated.jl

```diff
@@ -1565,6 +1562,14 @@ end
 @deprecate ucfirst uppercasefirst
 @deprecate lcfirst lowercasefirst
 
+function search(buf::IOBuffer, delim::UInt8)
```

Comment:
This should be for `GenericIOBuffer`.

Reply:
...with the more generic implementation of course. Otherwise it seems we're missing a deprecation for `search(::GenericIOBuffer, ...`

Reply:
`GenericIOBuffer` isn't exported, so is it really worth having a deprecation?